### PR TITLE
add buff/debuff arrays to artifacts

### PR DESCRIPTION
**Pull request by Kenford20**
[#171](https://github.com/EpicSevenDB/gamedatabase/issues/171)

used the keys I added in my other PR, tried to follow naming convention and used the ones provided in the buff-debuff-sheet. 

feel free to bring up any suggestions/concerns :)

### CODE REVIEW

**Comment by Kenford20** [comment]
just realized `stic_heal` key refers to continuous heal, not just any general heal, so I'll need to remove it from a handful of artifacts.

**Comment by Kenford20** [comment]
before I make the change above, there are 9 artifacts that have some sort of healing effect, while only one with continuous healing - forest totem. 

I think it'll be good to be able to filter out which artifacts provide ANY sort of healing effect (like continuous heals, one time heals, increased heal effect) vs solely continuous healing, from a users perspective. 

idk if this is something yall want to implement on the hero page as well, but there are certainly more heroes with continuous healing (also i feel its less necessary since heroes page has the 'soulweaver' filter)

let me know of your thoughts!

File: src/artifact/radiant-forever.json [hunk 1]
```diff
@@ -20,6 +20,10 @@
         "max": {
             "atk": 234,
             "hp": 351
-        }
+        },
+        "buffs": [
+            "stic_att_up"
```
Comment (kmalone86) [inline src/artifact/radiant-forever.json]:
```suggestion
            "stic_custom_dmg_up"
```

File: src/artifact/rainbow-scale.json [hunk 2]
```diff
@@ -19,5 +19,10 @@
             "atk": 156,
             "hp": 585
         }
-    }
+    },
+    "buffs": [
+        "stic_efctness_up",
```
Comment (kmalone86) [inline src/artifact/rainbow-scale.json]:
```suggestion
        "stic_custom_eff_up",
```

File: src/artifact/rainbow-scale.json [hunk 3]
```diff
@@ -19,5 +19,10 @@
...
+        "stic_evasion_up"
```
Comment (kmalone86) [inline src/artifact/rainbow-scale.json]:
```suggestion
        "stic_custom_efr_up"
```

File: src/artifact/reingars-special-drink.json [hunk 4]
```diff
@@ -20,5 +20,9 @@
             "atk": 273,
             "hp": 416
         }
-    }
+    },
+    "buffs": [
+        "stic_att_up"
```
Comment (kmalone86) [inline src/artifact/reingars-special-drink.json]:
```suggestion
        "stic_custom_dmg"
```

File: src/artifact/rod-of-amaryllis.json [hunk 5]
```diff
@@ -20,5 +20,9 @@
             "atk": 117,
             "hp": 988
         }
-    }
+    },
+    "buffs": [
+        "stic_heal"
```
Comment (kmalone86) [inline src/artifact/rod-of-amaryllis.json]:
```suggestion
        "stic_heal_hp"
```

File: src/artifact/rosa-hargana.json [hunk 6]
```diff
@@ -20,5 +20,9 @@
             "atk": 234,
             "hp": 351
         }
-    }
+    },
+    "buffs": [
+        "efct_dual_att"
```
Comment (kmalone86) [inline src/artifact/rosa-hargana.json]:
```suggestion
        "efct_dual_att",
        "stic_custom_att_up"
```

File: src/artifact/shepherds-of-chaos.json [hunk 7]
```diff
@@ -21,5 +21,10 @@
             "atk": 234,
             "hp": 351
         }
-    }
+    },
+    "buffs": [
+        "stic_evasion_up",
```
Comment (kmalone86) [inline src/artifact/shepherds-of-chaos.json]:
```suggestion
        "stic_custom_dodge_up",
```

File: src/artifact/shimadra-staff.json [hunk 8]
```diff
@@ -21,5 +21,9 @@
             "atk": 195,
             "hp": 702
         }
-    }
+    },
+    "buffs": [
+        "stic_heal"
```
Comment (kmalone86) [inline src/artifact/shimadra-staff.json]:
```suggestion
        "stic_max_heal_up"
```

File: src/artifact/sira-ren.json [hunk 9]
```diff
@@ -20,5 +20,9 @@
             "atk": 156,
             "hp": 585
         }
-    }
+    },
+    "buffs": [],
+    "debuffs": [
+        "stic_rnd_debuf"
```
Comment (kmalone86) [inline src/artifact/sira-ren.json]:
```suggestion
        "efct_rnd_debuf",
        "stic_att_dn",
        "stic_blind",
        "stic_silence",
        "stic_sleep",
        "stic_dot"
```

File: src/artifact/song-of-stars.json [hunk 10]
```diff
@@ -20,5 +20,10 @@
             "atk": 195,
             "hp": 702
         }
-    }
+    },
+    "buffs": [
+        "stic_sign",
+        "stic_att_up"
```
Comment (kmalone86) [inline src/artifact/song-of-stars.json]:
```suggestion
        "stic_custom_dmg_up"
```

File: src/artifact/spooky-solayu-stories.json [hunk 11]
```diff
@@ -19,5 +19,9 @@
             "atk": 169,
             "hp": 624
         }
-    }
+    },
+    "buffs": [
+        "stic_efctness_up"
```
Comment (kmalone86) [inline src/artifact/spooky-solayu-stories.json]:
```suggestion
        "stic_custom_eff_up"
```

File: src/artifact/staff-of-wisdom.json [hunk 12]
```diff
@@ -18,5 +18,9 @@
             "atk": 117,
             "hp": 416
         }
-    }
+    },
+    "buffs": [
+        "stic_efctness_up"
```
Comment (kmalone86) [inline src/artifact/staff-of-wisdom.json]:
```suggestion
        "stic_custom_eff_up"
```

File: src/artifact/strak-gauntlet.json [hunk 13]
```diff
@@ -21,5 +21,9 @@
             "atk": 91,
             "hp": 819
         }
-    }
+    },
+    "buffs": [
+        "stic_evasion_up"
```
Comment (kmalone86) [inline src/artifact/strak-gauntlet.json]:
```suggestion
        "stic_custom_efr_up"
```

File: src/artifact/sword-of-ezera.json [hunk 14]
```diff
@@ -22,5 +22,9 @@
             "atk": 117,
             "hp": 988
         }
-    }
+    },
+    "buffs": [
+        "stic_def_up"
```
Comment (kmalone86) [inline src/artifact/sword-of-ezera.json]:
```suggestion
        "stic_rcv_dmg_dn"
```

File: src/artifact/time-matter.json [hunk 15]
```diff
@@ -20,5 +20,10 @@
             "atk": 234,
             "hp": 559
         }
-    }
+    },
+    "buffs": [
+        "efct_cd_dn",
+        "stic_att_up"
```
Comment (kmalone86) [inline src/artifact/time-matter.json]:
```suggestion
        "stic_custom_dmg_up"
```

File: src/artifact/uberiuss-tooth.json [hunk 16]
```diff
@@ -20,5 +20,10 @@
             "atk": 273,
             "hp": 416
         }
-    }
+    },
+    "buffs": [
+        "stic_att_up",
+        "stic_cridmg_up"
```
Comment (kmalone86) [inline src/artifact/uberiuss-tooth.json]:
```suggestion
```

File: src/artifact/venus-orb.json [hunk 17]
```diff
@@ -18,5 +18,9 @@
             "atk": 91,
             "hp": 312
         }
-    }
+    },
+    "buffs": [
+        "stic_efctness_up"
```
Comment (kmalone86) [inline src/artifact/venus-orb.json]:
```suggestion
        "stic_custom_eff_up"
```

File: src/artifact/violet-talisman.json [hunk 18]
```diff
@@ -21,5 +21,10 @@
             "atk": 273,
             "hp": 416
         }
-    }
+    },
+    "buffs": [
+        "stic_att_up",
+        "stic_evasion_up"
```
Comment (kmalone86) [inline src/artifact/violet-talisman.json]:
```suggestion
        "stic_custom_dodge_up"
```

File: src/artifact/waters-origin.json [hunk 19]
```diff
@@ -21,5 +21,10 @@
             "atk": 91,
             "hp": 819
         }
-    }
+    },
+    "buffs": [
+        "stic_heal",
```
Comment (kmalone86) [inline src/artifact/waters-origin.json]:
```suggestion
        "stic_heal_hp",
```

**Comment by kmalone86** [comment]
Thanks for adding. I have added a bunch of comments.

@RaphaelDDL   I don't think we should have stic_att_up in artifacts as this would get confused with the buff, which increases attack 50%. I added as stic_custom_att_up, just to separate them.  Any thoughts on this Raph?

Aside from adding stic_custom-..., I think there are a few new buff/effects we need to take into account.
stic_cridmg_dn, stic_cridmg_up, stic_heal_hp, stic_rcv_dmg_dn, stic_dmg_up, stic_max_heal_up, stic_exp_up, stic_custom_dmg

**Comment by kmalone86** [comment]
Also I think tear of the desert was missed  stic_custom_att_up

**Comment by kmalone86** [comment]
Great github is blowing up due to too many change requests

File: src/artifact/uberiuss-tooth.json [hunk 20]
```diff
@@ -20,5 +20,10 @@
             "atk": 273,
             "hp": 416
         }
-    }
+    },
+    "buffs": [
+        "stic_att_up",
```
Comment (kmalone86) [inline src/artifact/uberiuss-tooth.json]:
```suggestion
        "stic_custom_att_up"
```

**Comment by RaphaelDDL** [comment]
> @RaphaelDDL I don't think we should have stic_att_up in artifacts as this would get confused with the buff, which increases attack 50%. I added as stic_custom_att_up, just to separate them. Any thoughts on this Raph?

On this point , I have mixed feelings. On a user perspective, they'll see on the filter "attack up", which the mentioned artifacts give. The issue is that each artifact might grant a different amount of attack increase, while, as you said, the real attack buff is 50%.

Data wise, it's wrong, but user wise, it's easier to have one single "attack up" filter rather than an "attack up", "random attack up" and etc etc. 

So when someone needs to apply a filter, they'll need to choose between atk up (buff) or attack buff (random) ?

**Comment by kmalone86** [comment]
@RaphaelDDL We could just filter on the one and show both buff and custom/random.
Might be easier to do this if it is stic_att_up & stic_att_up_custom, so you can just lookup stic_att_up and just accept if there is extra working after, but only show the buff icon if there is no _custom.

Don't forget there are some artifacts that do give the normal att_up buff that we would probably won't the icon displayed on the page.

**Comment by RaphaelDDL** [comment]
yeah, having by the end should be enough then, because then instead of `Array.includes('stic_att_up')` (which is specific to `stic_att_up`), I could do something to apply the filter to all a find or map which checks the  `String.startWith('stic_att_up')` which would give both `stic_att_up` and `stic_att_up_custom`

**Comment by Kenford20** [comment]
I was aware that the att_up buff refers to the actual 50% buff, but at the moment, I was probably thinking from a users perspective that filtering artifacts based on 'attack' should provide any artifact that gives some kind of attack bonus for those who just want to explore artifacts that will 'make their hero do more dmg', so I ultimately just labeled anything as such. 

but yeah I didn't have the context on how these keys were being used, so I will make changes based on the comment above!

**Comment by Kenford20** [comment]
I will also apply this convention for other stat buffs ie: custom defense increases, custom crit bonuses, etc

**Comment by Kenford20** [comment]
addressed changes!

**Pull request merged**